### PR TITLE
Fix Notifier::Base tests

### PR DESCRIPTION
**Pull request by xzhflying**

### CODE REVIEW

File: spec/notifiers/notifier/base_spec.rb [hunk 1]
```diff
@@ -3,88 +3,99 @@
...
+    end
+
+    context 'when a specific notifier is provided' do
+      let(:user) { create(:user) }
+      let(:course) { create(:course) }
+      let!(:course_users) { create(:course_user, course: course) }
+      let(:notifier) { self.class::DummyNotifier.new }
+      let(:template) { 'activity_mailer/test_email' }
+
+      context 'when notify a user' do
```
Comment (lowjoel) [inline spec/notifiers/notifier/base_spec.rb]: when notify*ing*

File: spec/notifiers/notifier/base_spec.rb [hunk 2]
```diff
@@ -3,88 +3,99 @@
...
-          before do
-            allow(notifier).to receive(:notification_view_path).and_return(template)
-          end
+        it 'sends an email notification' do
+          expect { subject }.to change { ActionMailer::Base.deliveries.count }.by(1)
+        end
+      end
 
-          subject { notifier.dummy_updated(user, user, course) }
+      context 'when notify a course' do
```
Comment (lowjoel) [inline spec/notifiers/notifier/base_spec.rb]: notifying

File: spec/notifiers/notifier/base_spec.rb [hunk 3]
```diff
@@ -3,88 +3,99 @@
...
+        it 'sends an email notification' do
+          expect { subject }.to change { ActionMailer::Base.deliveries.count }.by(2)
         end
       end
+    end
+
+    describe '.method_missing' do
+      subject { self.class::DummyNotifier.method_missing_tester }
+
+      it 'returns pass' do
```
Comment (lowjoel) [inline spec/notifiers/notifier/base_spec.rb]: it 'calls the instance method'

**Comment by xzhflying** [comment]
coverall says there is one line I do not cover in `base.rb`. But actually I've tested that line. Seems there is something wrong with coverall.

**Comment by lowjoel** [comment]
Nice!

**Pull request merged**